src/diffusers/pipelines/blip_diffusion/pipeline_blip_diffusion.py

```diff
@@ -0,0 +1,431 @@
+            raise ImportError("`enable_model_cpu_offload` requires `accelerate v0.17.0` or higher.")
+
+        device = torch.device(f"cuda:{gpu_id}")
+
+        if self.device.type != "cpu":
+            self.to("cpu", silence_dtype_warnings=True)
+            torch.cuda.empty_cache()  # otherwise we don't see the memory savings (but they probably exist)
+
+        hook = None
+        for cpu_offloaded_model in [self.unet, self.qformer, self.vae, self.text_encoder]:
+            _, hook = cpu_offload_with_hook(cpu_offloaded_model, device, prev_module_hook=hook)
+
+        # We'll offload the last model manually.
+        self.final_offload_hook = hook
+
+    @property
+    # Copied from diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion.StableDiffusionPipeline._execution_device
+    def _execution_device(self):
+        r"""
+        Returns the device on which the pipeline's models will be executed. After calling
+        `pipeline.enable_sequential_cpu_offload()` the execution device can only be inferred from Accelerate's module
+        hooks.
+        """
+        if not hasattr(self.unet, "_hf_hook"):
+            return self.device
+        for module in self.unet.modules():
+            if (
+                hasattr(module, "_hf_hook")
+                and hasattr(module._hf_hook, "execution_device")
+                and module._hf_hook.execution_device is not None
+            ):
+                return torch.device(module._hf_hook.execution_device)
+        return self.device
+
+
+    def _build_prompt(self, prompts, tgt_subjects, prompt_strength=1.0, prompt_reps=20):
+        rv = []
+        for prompt, tgt_subject in zip(prompts, tgt_subjects):
+            prompt = f"a {tgt_subject} {prompt.strip()}"
+            # a trick to amplify the prompt
+            rv.append(", ".join([prompt] * int(prompt_strength * prompt_reps)))
+
+        return rv
+
+    def _predict_noise(
+        self,
+        t,
+        latent_model_input,
+        text_embeddings,
+        width=512,
+        height=512,
+        cond_image=None,
+    ):
+ 
+        down_block_res_samples, mid_block_res_sample = None, None
+
+        noise_pred = self.unet(
+            latent_model_input,
+            timestep=t,
+            encoder_hidden_states=text_embeddings,
+            down_block_additional_residuals=down_block_res_samples,
+            mid_block_additional_residual=mid_block_res_sample,
+        )["sample"]
+
+        return noise_pred
+
+    def _init_latent(self, latent, height, width, generator, batch_size):
+        if latent is None:
+            latent = torch.randn(
+                (1, self.unet.in_channels, height // 8, width // 8),
+                generator=generator,
+                device=generator.device,
+            )
+        latent = latent.expand(
+            batch_size,
+            self.unet.in_channels,
+            height // 8,
+            width // 8,
+        )
+        return latent.to(self.device)
```

Comment:
We make use of 

https://github.com/huggingface/diffusers/blob/078df46bc9a99178a9a744b872899990353769a4/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L502